### ISSUE TITLE
Single context for the connected client at server side

### ISSUE THREAD

**Issue by biovoid** [dotnet/MQTTnet#606]
Hi,
I am using a single server class with all interface to handle the client operations:
```
IMqttServerConnectionValidator
IMqttServerSubscriptionInterceptor
IMqttServerApplicationMessageInterceptor
IMqttServerClientConnectedHandler
IMqttServerClientDisconnectedHandler
```
And every callback I am searching the client entity by ClientId. May be it will be better to add save context from user code at ValidateConnectionAsync and let user access it on every callback without resolving single item every callback.
For example, at least, I can save a logger item for specific client and use this logger only for this client without code like this every time:
```
private static Logger GetLogger(string name)
{
     lock (locker)
     {
          if (String.IsNullOrEmpty(name))
              return LogManager.GetCurrentClassLogger();
          else
              return LogManager.GetLogger(name);
     }
}

...

public async Task InterceptSubscriptionAsync(MqttSubscriptionInterceptorContext context)
{
     Logger logger = GetLogger(context.ClientId);
     ...
}

**Comment by chkr1011** [comment]
Hi,
you can make use of the Session items. These are shared in all handler and allow keeping custom data. These are available in 3.0.6+.

So you can add the logger as a value to the session items.

Best regards
Christian

**Comment by SeppPenner** [comment]
I guess, this is answered already. If not, feel free to re-open the issue.
